### ISSUE TITLE
[Segment Replication] Update replica recovery to perform replication from source

### ISSUE THREAD

**Issue by dreamer-89** [opensearch-project/OpenSearch#5743]
Coming from https://github.com/opensearch-project/OpenSearch/pull/5722#discussion_r1063671145, the existing replica recovery performs a round of segment replication when recovery is marked complete on target. This also introduced dependency on SegmentReplicationTargetService inside IndicesClusterStateService. This is not needed as same can be achieved by invoking ForceSegmentReplication handler inside SegmentReplicationTargetService.

**Comment by dreamer-89** [comment]
Opened a PR, but observing unit test failures which needs deep dive. Will look into it after #5848

**Comment by dreamer-89** [comment]
I looked into the test failure `testNRTReplicaPromotedAsPrimary` and seems to be happening due to stale translog generation which results in all operations been available on translog.
I see it was passing previously as replica were started with -1 for `lastReceivedGen`. While primary post indexing has higher generation for SegmentInfos which results in new `TranslogWriter` on repilca (resulting in wiping out of previous operations). So, below assertions were not tripping. With my change, which performs SegRep during reocvery bring the replica’s generation to the primary’s. During indexing and round of segrep, replica doesn’t receive higher SegmentInfos.version which results in writes to existing TranslogWriter; with end result of all documents been present on current TranslogWriter rather than the ones, ingested post SegRep.

```
assertEquals(additonalDocs, nextPrimary.translogStats().estimatedNumberOfOperations());
assertEquals(additonalDocs, replica.translogStats().estimatedNumberOfOperations());
assertEquals(additonalDocs, nextPrimary.translogStats().getUncommittedOperations());
assertEquals(additonalDocs, replica.translogStats().getUncommittedOperations());
```

**Comment by dreamer-89** [comment]
Closing this issue as change is merged into `main` in https://github.com/opensearch-project/OpenSearch/pull/5746 and backported in https://github.com/opensearch-project/OpenSearch/pull/6149
